layers/+lang/haskell/README.org

```diff
@@ -116,28 +134,48 @@ Backend can be chosen on a per project basis using directory local variables
 
 *Note:* you can easily add a directory local variable with ~SPC f v d~.
 
+There are two backends available: =dante= and =lsp=.
+
+=dante= is lightweight, requires no setup and works out of the box in most cases, which is why it is also a default backend.
+
+=lsp= (=hie= or =hls=) is a more ambitious, heavy-weight, cutting-edge backend that is however still somewhat rough on the edges and requires some additional setup.
+
 *** =dante=
-[[https://github.com/jyp/dante][Dante]] is a fork of Intero mode which aims exclusively at providing a convenient frontend to GHCi.
+[[https://github.com/jyp/dante][Dante]] is a lightweight backend which delegates most of its work directly to GHCi.
 
 It brings features like syntax checking, auto completion, hlint suggestions, automatic error fixing, info at point, definition and use sites.
 
 =dante= works for =cabal=, =nix=, =sytx=, and =stack= users and requires no additional setup.
 
-=dante= requires Emacs 25.
-
 *** =lsp=
-=lsp= requires an appropriate installation of =hie=, the official Haskell language server, to work properly.
+[[https://microsoft.github.io/language-server-protocol][Language Server Protocol]] is a standard for implementing language backends.
+
+In Haskell layer, you can use a backend that implements Language Server Protocol for Haskell by specifying =lsp= as backend
+and then installing concrete backend implementation, of which there are two available at the moment: =hie= and =hls=.
+
+Enabling the =lsp= backend requires the =lsp= layer to be enabled, and provides access to
+all the additional =lsp-mode= key bindings. =lsp= uses =lsp-ui= to show the errors/warnings instead of =flycheck=. 
+
+**** [[https://github.com/haskell/haskell-ide-engine][Haskell Ide Engine]] (=hie=)
```

Comment:
Can you verify that links as headlines work in org and markdown? I want to avoid issues with the document conversion beforehand.

Reply:
I can confirm that they work!

Reply:
This is already happening, for the spacemacs websites, especially for develop.spacemacs.org, for this org is converted to markdown. You can check if this still works by running `spacemacs/publish-doc` in `core/core-documentation.el`. This should not fail for the haskell layer.

Reply:
Ok makes sense! I actually just moved links to the text, instead of headers, I think it is just eaiser that way. I did run spacemacs/publish-doc from core/core-documentation.el, but after I moved the links from headers, and it generated a lot of stuff and I haven't seen any errors reported. However maybe I missed some error there was a lot of logs, and I wasn't sure where docs got generated so I didn't inspect them. Anyway this should solve it!